src/industry_cmd.cpp

```diff
@@ -1413,21 +1417,21 @@ bool IsSlopeRefused(Slope current, Slope refused)
 /**
  * Are the tiles of the industry free?
  * @param tile                    Position to check.
- * @param it                      Industry tiles table.
- * @param itspec_index            The index of the itsepc to build/fund
+ * @param layout                  Industry tiles table.
+ * @param layout_index            The index of the layout to build/fund
  * @param type                    Type of the industry.
  * @param initial_random_bits     The random bits the industry is going to have after construction.
  * @param founder                 Industry founder
  * @param creation_type           The circumstances the industry is created under.
  * @param[out] custom_shape_check Perform custom check for the site.
  * @return Failed or succeeded command.
  */
-static CommandCost CheckIfIndustryTilesAreFree(TileIndex tile, const IndustryTileTable *it, uint itspec_index, int type, uint16 initial_random_bits, Owner founder, IndustryAvailabilityCallType creation_type, bool *custom_shape_check = nullptr)
+static CommandCost CheckIfIndustryTilesAreFree(TileIndex tile, const IndustryTileLayout &layout, uint itspec_index, int type, uint16 initial_random_bits, Owner founder, IndustryAvailabilityCallType creation_type, bool *custom_shape_check = nullptr)
 {
 	bool refused_slope = false;
 	bool custom_shape = false;
 
-	do {
+	for (IndustryTileLayout::const_iterator it = layout.begin(); it != layout.end(); ++it) {
```

Comment:
Several places where range for could be used, I think?

Reply:
It could, I used an explicit iterator to keep the body of the loop unchanged.